joint_state_publisher/README.md

```diff
@@ -25,5 +25,10 @@ Parameters
 * `use_smallest_joint_limits` (bool) - Whether to honor `<safety_controller>` tags in the URDF.  Defaults to True.
 * `source_list` (array of strings) - Each string in this array represents a topic name.  For each string, create a subscription to the named topic of type `sensor_msgs/msg/JointStates`.  Publication to that topic will update the joints named in the message.  Defaults to an empty array.
 * `delta` (double) - How much to automatically move joints during each iteration.  Defaults to 0.0.
-* `zeros` (dictionary of string -> float) - A dictionary of joint_names to initial starting values for the joint.  In Eloquent, pass on the command-line as '-p zeros.joint_name:=value'.  Defaults to an empty dictionary, in which case 0 is assumed as the zero for all joints.
-* `dependent_joints` (dictionary of string -> dictionary of 'parent', 'factor', 'offset') - A dictionary of joint_names to the joints that they mimic; compare to the `<mimic>` tag in URDF.  A joint listed here will mimic the movements of the 'parent' joint, subject to the 'factor' and 'offset' provided.  The 'parent' name must be provided, while the 'factor' and 'offset' parameters are optional (they default to 1.0 and 0.0, respectively).  In Eloquent, pass on the command-line as '-p dependent_joints.left_leg.parent:=right_leg -p dependent_joints.left_leg.offset:=0.0 -p dependent_joints.left_leg.factor:=2.0'. Defaults to the empty dictionary, in which case only joints that are marked as `<mimic>` in the URDF are mimiced.
+
+#### Mapped Parameters
+
+These parameters won't show up under '/joint_state_publisher' with `ros2 param list`, because they are declared differently than the other parameters. For a dictionary, the format to use these parameters is `<parameter>.<key>:=<value>`, where a new parameter is defined for each key. See below for examples.
```

Comment:
So this isn't totally true, or at least it is not meant to be true.  If you just run:

```
ros2 run joint_state_publisher joint_state_publisher
```

it is true that there will be no declared `zero.<joint>` parameters.  But that's because we don't know what the joints might be.

Instead, if you run:

```
ros2 run joint_state_publisher joint_state_publisher --ros-args -p zeros.joint2:=3.0
```

Then the `zeros.joint2` parameter will indeed show up in `ros2 param list /joint_state_publisher`.

Theoretically, at the end of [configure_robot](https://github.com/ros/joint_state_publisher/blob/ros2/joint_state_publisher/joint_state_publisher/joint_state_publisher.py#L147), we could do a `declare_parameter('zeros.<jointname>', 0.0)` for every possible joint that we see in the incoming URDF.  But that would mean that on a large robot we would have hundreds of parameters, most of which will never be used.  I'm not sure if that is worth it or not.

Reply:
Agreed, that phrase wasn't entirely true. To make it unambiguous that these parameters exist, I've added a commit that declares the `zeros` and `dependent_joints` parameters as NOT_SET. The node prints warnings when attempting to interact with these parameters directly instead of using `<parameter>.<key>`. The README removes the not true statement.

This has the benefit of being able to explore the node's parameters using the cli. In the future, these parameters could have descriptions.

I agree that it's likely not worth declaring every parameter from the URDF; the large printout would make it harder to read the rest of the parameters.

If you think this change doesn't make sense, let me know and I can revert it.